### PR TITLE
add support for EF signing

### PR DESCRIPTION
**Pull request by gdams**

### CODE REVIEW

**Comment by gdams** [comment]
Sample macOS installer:

Screenshot 2021-05-14 at 12 51 11

**Comment by karianna** [comment]
@gdams conflicts

File: codesign/eclipse-codesign.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+set -eu
+
+VENDOR="adoptium"
```
Comment (karianna) [inline codesign/eclipse-codesign.sh:19]: Is that correct?  Not Eclipse Adoptium?

File: codesign/eclipse-notarize.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+set -eu
+
+PKG="$1"
+PRIMARY_BUNDLE_ID="$2"
+
+echo "Notarizing $1, this can take a while!"
```
Comment (karianna) [inline codesign/eclipse-notarize.sh:22]: Might be worth adding that you'll update status every minute

File: codesign/eclipse-notarize.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+while [[ ${STATUS} == 'IN_PROGRESS' ]]; do
+    sleep 1m
+    RESPONSE=$(curl -s https://cbi.eclipse.org/macos/xcrun/${UUID}/status)
+    STATUS=$(echo $RESPONSE | grep -Po '"status"\s*:\s*"\K[^"]+')
+done
+
+if [[ ${STATUS} != 'COMPLETE' ]]; then
+    echo "Notarization failed: ${RESPONSE}"
+    exit 1
+fi
```
Comment (karianna) [inline codesign/eclipse-notarize.sh:38]: does cleanup need to occur in the failure case?

File: pkgbuild/packagesbuild.sh [hunk 4]
```diff
@@ -162,7 +162,7 @@ cp -R "${INPUT_DIRECTORY}" "${DIRECTORY}"
 if [ ! -z "$SIGN_OPTION" ]; then
     xattr -cr .
     security unlock-keychain -p `cat ~/.password` login.keychain-db
-    /usr/bin/codesign --verbose=4 --deep --force -s "Developer ID Application: London Jamocha Community CIC" ${DIRECTORY}
+    /usr/bin/codesign --verbose=4 --deep --force -s - ${DIRECTORY}
```
Comment (karianna) [inline pkgbuild/packagesbuild.sh:165]: is the `-` correct?

**Review by karianna — CHANGES_REQUESTED** [review]
Some nitpicks

File: codesign/eclipse-codesign.sh (hunk 1 above)
Comment (gdams) [reply codesign/eclipse-codesign.sh:19]: this is correct, it defines the jdk identifier which is `net.temurin.11.jdk`

File: codesign/eclipse-notarize.sh (hunk 3 above)
Comment (gdams) [reply codesign/eclipse-notarize.sh:38]: the workspace gets cleaned up each run anyway

File: pkgbuild/packagesbuild.sh (hunk 4 above)
Comment (gdams) [reply pkgbuild/packagesbuild.sh:165]: it's a template, it allows other users to add their cert (see the comment a few lines up)

File: codesign/eclipse-notarize.sh (hunk 2 above)
Comment (gdams) [reply codesign/eclipse-notarize.sh]: done

File: codesign/eclipse-notarize.sh [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+set -eu
+
+PKG="$1"
+PRIMARY_BUNDLE_ID="$2"
```
Comment (tellison) [inline codesign/eclipse-notarize.sh:20]: Consider validating parameters, to ensure they are actually passed in before proceeding.

File: codesign/eclipse-notarize.sh [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+done
+
+if [[ ${STATUS} != 'COMPLETE' ]]; then
+    echo "Notarization failed: ${RESPONSE}"
+    exit 1
+fi
+
+rm "${PKG}"
+
+curl -JO https://cbi.eclipse.org/macos/xcrun/${UUID}/download
```
Comment (tellison) [inline codesign/eclipse-notarize.sh:42]: This brings back the file named ${PKG} ? Consider making that explicit.

**Review by tellison — APPROVED** [review]
Minor comments

**Review by karianna — APPROVED** [review]

File: codesign/eclipse-notarize.sh (hunk 5 above)
Comment (gdams) [reply codesign/eclipse-notarize.sh:20]: `set -eu` should cause this to fail if they are undefined

**Pull request merged**